### PR TITLE
Implement BananaStatusBarController

### PR DESCRIPTION
**Pull request by jdragonbae**
This patch implements BananaStatusBarController to set color on the
status bar. It will set the color of status bar when the secure dns mode
is on. Also, this helps to inform the user that the feature is working.

Refs: #1082

### CODE REVIEW

File: src/org/triple/banana/util/StatusBarController.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        final Activity activity = BananaApplicationUtils.get().getLastTrackedFocusedActivity();
+        if (activity == null) return;
+        final View root = activity.getWindow().getDecorView().getRootView();
+        BananaStatusBarController.setStatusBarColor(activity.getWindow(), color);
+        BananaStatusBarController.setStatusBarIconColor(root, useDarkIcons);
+    }
+
+    private @ColorInt int getColorFromResource(@ColorRes int res) {
+        return ContextCompat.getColor(BananaApplicationUtils.get().getApplicationContext(), res);
+    }
```
Comment (romandev) [inline src/org/triple/banana/util/StatusBarController.java:54]:
```
@Override
public @ColorInt int overrideStatusBarColor(@ColorInt int color) {
    if (!BananaApplicationUtils.get().isNativeInitialzed() || !BananaSecureDnsBridge.get().isSecure()) return color;
    return ContextCompat.getColor(BananaApplicationUtils.get().getApplicationContext(), R.color.secure_dns_status_bar_color);
}
```

File: banana_cake/interfaces/java/org/banana/cake/interfaces/BananaStatusBarController.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    }
+    static void setStatusBarColor(@NonNull Window window, @ColorInt int color) {
+        ApiCompatibilityUtils.setStatusBarColor(window, color);
+    }
+    static void setStatusBarIconColor(@NonNull View view, boolean useDarkIcons) {
+        ApiCompatibilityUtils.setStatusBarIconColor(view, useDarkIcons);
+    }
+
+    boolean setStatusBarColorIfNeeded();
+    void onInitialized();
```
[the remark this answers is not in the document]
Comment (jdragonbae) [reply banana_cake/interfaces/java/org/banana/cake/interfaces/BananaStatusBarController.java]: Thanks for the review. Done.

File: src/org/triple/banana/util/StatusBarController.java (hunk 1 above)
Comment (jdragonbae) [reply src/org/triple/banana/util/StatusBarController.java]: Done

File: banana_cake/java/org/banana/cake/CakeApplicationUtils.java [hunk 3]
```diff
@@ -83,4 +84,9 @@ public void openBrowserSettings(@NonNull Context context) {
     public @Nullable Activity getLastTrackedFocusedActivity() {
         return ApplicationStatus.getLastTrackedFocusedActivity();
     }
+
+    @Override
+    public boolean isNativeInitialized() {
+        return LibraryLoader.getInstance().isInitialized();
```
Comment (romandev) [inline banana_cake/java/org/banana/cake/CakeApplicationUtils.java:90]: Does this work well?

Comment (jdragonbae) [reply banana_cake/java/org/banana/cake/CakeApplicationUtils.java:90]: In my device (P OS), it seems to be worked well. 
Is there any reasons for problems?

Comment (romandev) [reply banana_cake/java/org/banana/cake/CakeApplicationUtils.java:90]: I think it should be `FeatureList.isNativeInitialized();`

File: res/values/colors.xml [hunk 4]
```diff
@@ -21,4 +21,5 @@
     <color name="toolbar_modern_grey_background_color">#202124</color>
     <color name="toolbar_modern_grey_text_color">#3C4043</color>
     <color name="toolbar_button_diabled">#F2F2F2</color>
+    <color name="secure_dns_status_bar_color">#7c4dff</color>
```
Comment (romandev) [inline res/values/colors.xml:24]: Should be `#571CCE`

File: banana_cake/java/org/banana/cake/CakeApplicationUtils.java (hunk 3 above)
Comment (jdragonbae) [reply banana_cake/java/org/banana/cake/CakeApplicationUtils.java]: Thanks. Done.

File: res/values/colors.xml (hunk 4 above)
Comment (jdragonbae) [reply res/values/colors.xml]: Thanks. Done.

**Review by romandev — APPROVED** [review]
lgtm

**Pull request merged**